src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,538 @@
+		// intermediateData.rooms.length = Math.min(intermediateData.rooms.length, 3);
+		/* TODO: remove end */
+
+		// Iteration approaches currently implemented: teachers and rooms
+		if (intermediateData.teachers !== undefined) { // Iterate over teachers
+			for (const teacher of intermediateData.teachers) {
+				const name = `${teacher.foreName} ${teacher.longName}`;
+
+				let timetable = await this.getCustomizableTimeTableFor(2, teacher.id, {
+					startDate: intermediateData.currentSchoolYear.startDate,
+					endDate: intermediateData.currentSchoolYear.endDate,
+					onlyBaseTimetable: true,
+					klasseFields: ['id', 'longname'],
+					subjectFields: ['id', 'longname'],
+					roomFields: ['id', 'longname'],
+				});
+
+				/* TODO: Check for change */
+				timetable = timetable.filter(entry => entry.te.length === 1
+					&& entry.te[0].id === teacher.id
+					&& entry.kl.length > 0
+					&& entry.ro.length === 1
+					&& entry.su.length === 1);
+				/* END TODO: Check for change */
+	
+				for (const entry of timetable) {
+					for (const classEntry of entry.kl) {
+						const klass = intermediateData.classes.find(k => k.id === classEntry.id);
+	
+						if (klass !== undefined) {
+							klass.timetable.push({
+								date: entry.date,
+								startTime: entry.startTime,
+								endTime: entry.endTime,
+								teacher: name,
+								subject: entry.su[0].longname,
+								room: entry.ro[0].longname,
+							});
+						}
+					}
+				}
+
+				// One teacher for now
+				// break;
+			}
+		} else { // Iterate over rooms
+			for (const room of intermediateData.rooms) {
+				const name = `${room.name} (${room.longName}${room.building !== '' ? `, ${room.building}` : ''})`;
+
+				let timetable = await this.getCustomizableTimeTableFor(4, room.id, {
+					startDate: intermediateData.currentSchoolYear.startDate,
+					endDate: intermediateData.currentSchoolYear.endDate,
+					onlyBaseTimetable: true,
+					klasseFields: ['id', 'longname'],
+					subjectFields: ['id', 'longname'],
+					teacherFields: ['id', 'longname'],
+				});
+
+				/* TODO: Check for change */
+				timetable = timetable.filter(entry => entry.ro.length === 1
+					&& entry.ro[0].id === room.id
+					&& entry.kl.length > 0
+					&& entry.te.length === 1
+					&& entry.su.length === 1);
+				/* END TODO: Check for change */
+	
+				for (const entry of timetable) {
+					for (const classEntry of entry.kl) {
+						const klass = intermediateData.classes.find(k => k.id === classEntry.id);
+	
+						if (klass !== undefined) {
+							klass.timetable.push({
+								date: entry.date,
+								startTime: entry.startTime,
+								endTime: entry.endTime,
+								teacher: entry.te[0].longname,
+								subject: entry.su[0].longname,
+								room: name,
+							});
+						}
+					}
+				}
+
+				// One room for now
+				// break;
+			}
+		}
+
+		// Collect times for each class
+		for (let klass of intermediateData.classes) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['klass' is never reassigned. Use 'const' instead.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)